src/main/java/de/tum/in/www1/artemis/service/exam/ExamImportService.java

```diff
@@ -0,0 +1,322 @@
+                    // We do not want to abort the whole exam import process, we only skip the relevant exercise
+                    if (optionalOriginalQuizExercise.isEmpty()) {
+                        break;
+                    }
+                    exerciseCopied = quizExerciseImportService.importQuizExercise(optionalOriginalQuizExercise.get(), (QuizExercise) exerciseToCopy);
+                }
+
+            }
+            // Attach the newly created Exercise to the new Exercise Group
+            if (exerciseCopied != null) {
+                exerciseGroupCopied.addExercise(exerciseCopied);
+            }
+        });
+        exerciseGroupRepository.save(exerciseGroupCopied);
+    }
+
+    /**
+     * Prepares a Programming Exercise for the import by setting irrelevant data to null.
+     * Additionally, the grading Criteria is loaded and attached to the exercise, as this needs to be released before the import
+     *
+     * @param newExercise      The new exercise which should be prepared for the import
+     */
+    private void prepareProgrammingExerciseForExamImport(final ProgrammingExercise newExercise) {
+
+        // we do not support the following values as part of exam exercises
+        newExercise.setBuildAndTestStudentSubmissionsAfterDueDate(null);
+        newExercise.setSubmissionPolicy(null);
+        newExercise.setReleaseDate(null);
+        newExercise.setDueDate(null);
+        newExercise.setAssessmentDueDate(null);
+        newExercise.setExampleSolutionPublicationDate(null);
+
+        // TODO: explain why this is actually necessary in this case (we don't do it in the "normal" single import)
+
+        // Fetch grading criterion into exercise
+        List<GradingCriterion> gradingCriteria = gradingCriterionRepository.findByExerciseIdWithEagerGradingCriteria(newExercise.getId());
+        newExercise.setGradingCriteria(gradingCriteria);
+
+        newExercise.forceNewProjectKey();
+    }
+
+    /**
+     * Creates a new Exercise Group with the title and boolean:isMandatory of the provided Exercise Group
+     *
+     * @param exerciseGroupToCopy Exercise Group, which title and isMandatory should be copied
+     * @param targetExam          The Exam to which the new exerciseGroup should be linked
+     * @return a new Exercise Group with the same title and isMandatory
+     */
+    private ExerciseGroup copyExerciseGroupWithTitleAndIsMandatory(ExerciseGroup exerciseGroupToCopy, Exam targetExam) {
+        ExerciseGroup exerciseGroupCopied = new ExerciseGroup();
+        exerciseGroupCopied.setTitle(exerciseGroupToCopy.getTitle());
+        exerciseGroupCopied.setIsMandatory(exerciseGroupToCopy.getIsMandatory());
+
+        // The Exam needs to be reloaded, so the changes within the last exercise group in {@link ExamImportService#copySingleExerciseGroupWithExercises} are updated
+        targetExam = examRepository.findWithExerciseGroupsAndExercisesByIdOrElseThrow(targetExam.getId());
```

Comment:
```suggestion
        var savedTargetExam = examRepository.findWithExerciseGroupsAndExercisesByIdOrElseThrow(targetExam.getId());
```

Codacy warns and I think that's reasonable. (Then use `savedTargetExam` instead of `targetExam`).